### PR TITLE
1401787: Use json file for caching pool type.

### PR DESCRIPTION
**Pull request by jirihnidek**
* When server is not available, then sub-man tries to read
  pool type from json file: /var/lib/rhsm/cache/pool_status.json
* Added another cache PoolStatusCache (subtype of StatusCache).
* Bug fix of: https://bugzilla.redhat.com/show_bug.cgi?id=1401787

### CODE REVIEW

File: src/subscription_manager/cache.py [hunk 1]
```diff
@@ -538,14 +554,10 @@ def requires_update(self):
...
         if self.identity.is_valid():
-            cp = self.cp_provider.get_consumer_auth_cp()
-            entitlement_list = []
-            try:
-                entitlement_list = cp.getEntitlementList(self.identity.uuid)
-            except Exception, e:
-                # In this case, return an empty map.  We just won't populate the field
-                log.debug('Problem attmepting to get entitlements from the server')
-                log.debug(e)
+            pool_cache = inj.require(inj.POOL_STATUS_CACHE)
```
Comment (kahowell) [inline src/subscription_manager/cache.py]: Please do this in `__init__` instead.

File: src/subscription_manager/cache.py [hunk 2]
```diff
@@ -513,9 +513,25 @@ def _sync_with_server(self, uep, consumer_uuid):
...
+    def _sync_with_server(self, uep, uuid):
+        self.server_status = uep.getEntitlementList(uuid)
+
+
 class PoolTypeCache(object):
+    """
+    Cache type of pool
+    """
+
+    # CACHE_FILE = "/var/lib/rhsm/cache/pool_type.json"
```
Comment (kahowell) [inline src/subscription_manager/cache.py]: This looks like commented code. Please remove.

**Comment by jirihnidek** [comment]
I will try to add unit test for this bug fix too. Should I create another branch and PR for it or is it OK to add it to this PR.

**Comment by jirihnidek** [comment]
[BUG 1401787: Use json file for caching pool type](https://trello.com/c/EsYE0twj/377-bug-1401787-use-json-file-for-caching-pool-type)

**Comment by kahowell** [comment]
@jirihnidek, please add the unit test to this branch (same PR). I usually just amend the commit, but alternatively, you're free to push another commit, but then we'll need to squash before merging.

**Pull request merged**